general/releases/2.8.md

```diff
@@ -0,0 +1,244 @@
+- [MSA-14-0042](https://moodle.org/mod/forum/discuss.php?d=275158) Lack of access check in IP lookup functionality
+- [MSA-14-0043](https://moodle.org/mod/forum/discuss.php?d=275159) Lack of group check in web service for Forum
+- [MSA-14-0044](https://moodle.org/mod/forum/discuss.php?d=275160) Hardware path disclosed in the error message
+- [MSA-14-0045](https://moodle.org/mod/forum/discuss.php?d=275161) XSS file upload possible through web service
+- [MSA-14-0046](https://moodle.org/mod/forum/discuss.php?d=275162) CSRF in LTI module
+- [MSA-14-0047](https://moodle.org/mod/forum/discuss.php?d=275163) Possible data loss in Wiki activity
+- [MSA-14-0048](https://moodle.org/mod/forum/discuss.php?d=275164) CSRF in forum tracking toggle
+- [MSA-14-0049](https://moodle.org/mod/forum/discuss.php?d=275165) Possible to print arbitrary message to user by modifying URL
+
+## For Developers
+
+- [MDL-47194](https://tracker.moodle.org/browse/MDL-47194) - Inbound email (such as already implemented for forums and files) can now easily be implemented by any plugin
+- [MDL-43835](https://tracker.moodle.org/browse/MDL-43835) - Unit tests on posgres and mysql use random sequence values
+- [MDL-46585](https://tracker.moodle.org/browse/MDL-46585) - [Strings can now be deprecated](/general/projects/api/string-deprecation#how-to-deprecate-a-string-) to help translators
+- [MDL-39840](https://tracker.moodle.org/browse/MDL-39840) - Plugins can have their personal environment checks
+- [MDL-41663](https://tracker.moodle.org/browse/MDL-41663) - Plugins renderers and renderables can be located in namespaces and autoloaded
+- [MDL-46207](https://tracker.moodle.org/browse/MDL-46207) - Scheduled tasks can control if they should run when plugin is disabled
+- [MDL-45513](https://tracker.moodle.org/browse/MDL-45513) - Unit tests can be executed against custom cache stores
+
+### API changes
+
+- There are new convenience functions to obtain <tt>$cm</tt> objects that belong to the <tt>cm_info</tt> class. <tt>get_course_and_cm_from_cmid($cmorid, $modulename)</tt> and <tt>get_course_and_cm_from_instance($instanceorid, $modulename)</tt> will efficiently obtain <tt>$course</tt> and <tt>$cm</tt> at the start of a module script in one line. Also, the new function <tt>cm_info::create($cm)</tt> can be used when you need a <tt>cm_info</tt> object, but only have a standard database record.
+- Choice event definitions have changed in [MDL-3034](https://tracker.moodle.org/browse/MDL-3034) [placeholder for this to be expanded]
+- [MDL-47496](https://tracker.moodle.org/browse/MDL-47496) - The unused lib/pear/XML/ library was removed.
+- [MDL-36014](https://tracker.moodle.org/browse/MDL-36014) - Changes to functions retrieving list of available cohorts
+- [MDL-28484](https://tracker.moodle.org/browse/MDL-28484) - New function is_https() that detects if page is served on https
+
+### Upgrade notes for developers
```

Comment:
The content in this section should be displayed as a list (right now, as there are no empty lines between then, they are displayed all together in the same line).